telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java

```diff
@@ -172,6 +197,14 @@ private void backoff(
   public void shutdown() {
     LOG.info("Shutting down the TelemetryClient background Executor");
     executor.shutdown();
+    try {
+      if (!executor.awaitTermination(shutdownSeconds, TimeUnit.SECONDS)) {
+        LOG.warn("couldn't shutdown within timeout");
+        executor.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+        LOG.error("interrupted graceful shutdown", e);
```

Comment:
Should call Thread.currentThread().interrupt() inside the catch to reset the flag.